### ISSUE TITLE
Add Accessibility for Button Inside Combobox

### ISSUE THREAD

**Issue by eforbes27** [Esri/calcite-design-system#10701]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

When adding a `calcite-button` inside a `calcite-combobox`, the button is not accessible to keyboards. It is also not read out by screen readers on mouse hover until first clicked. 

### Acceptance Criteria

Desired behavior would be for button to be accessible to keyboards and read out by screen readers on first mouse hover.

### Relevant Info

Wanted for Workflow Manager web app

### Which Component

calcite-combobox, calcite-button

### Example Use Case

From calcite doc page code pen

<calcite-combobox placeholder="Select a field">
        <calcite-combobox-item value="Natural Resources" text-label="Natural Resources"></calcite-combobox-item>
        <calcite-combobox-item value="Agriculture" text-label="Agriculture"></calcite-combobox-item>
        <calcite-combobox-item value="Forestry" text-label="Forestry"></calcite-combobox-item>
        <calcite-combobox-item value="Mining" text-label="Mining"></calcite-combobox-item>
        <calcite-combobox-item value="Business" text-label="Business"></calcite-combobox-item>
        <calcite-combobox-item value="Education" text-label="Education"></calcite-combobox-item>
        <calcite-combobox-item value="Utilities" text-label="Utilities"></calcite-combobox-item>
        <calcite-combobox-item value="Transportation" text-label="Transportation"></calcite-combobox-item>
      
      <calcite-button>Test Button 2</calcite-button>
    </calcite-combobox>

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by geospatialem** [comment]
Seeking design expertise on the above workflow for consideration.

**Comment by ashetland** [comment]
Blocked by #8707 

cc @geospatialem

**Comment by ashetland** [comment]
@Esri/calcite-pes This is labeled as `p-high`, but the issue blocking it, #8707, is a `p-low`. Should probably reassess the priorities on these two.
